src/main/java/fr/esgi/al5_2/Tayarim/controllers/ProprietaireController.java

```diff
@@ -22,144 +21,254 @@
+    authService.verifyToken(getTokenFromHeader(authHeader), true);
 
-    @GetMapping("/{id}")
-    public ResponseEntity<ProprietaireDTO> getProprietaire(@RequestHeader("Authorization") String authHeader, @PathVariable Long id, @RequestParam(name = "logement", defaultValue = "false") Boolean isLogement){
+    return new ResponseEntity<>(
+        proprietaireService.getProprietaire(isLogement),
+        HttpStatus.OK
+    );
+  }
 
-        authService.verifyToken(getTokenFromHeader(authHeader));
-
-        return new ResponseEntity<>(
-                proprietaireService.getProprietaireById(id, isLogement),
-                HttpStatus.OK
-        );
-    }
+  /**
+   * Obtient un propriétaire par son identifiant.
+   *
+   * @param authHeader L'en-tête d'autorisation contenant le token JWT.
+   * @param id         L'identifiant du propriétaire.
+   * @param isLogement Indicateur si le propriétaire retourné doit inclure des informations de
+   *                   logement.
+   * @return Une ResponseEntity contenant les détails du propriétaire et le statut HTTP.
+   */
+  @GetMapping("/{id}")
+  public ResponseEntity<ProprietaireDTO> getProprietaire(
+      @RequestHeader("Authorization") String authHeader, @PathVariable Long id,
+      @RequestParam(name = "logement", defaultValue = "false") Boolean isLogement) {
 
-    @PutMapping("/{id}")
-    public ResponseEntity<ProprietaireDTO> updateProprietaire(@RequestHeader("Authorization") String authHeader, @PathVariable Long id, @RequestBody ProprietaireUpdateDTO proprietaireUpdateDTO){
-        Long idToken = authService.verifyToken(getTokenFromHeader(authHeader)).getId();
+    authService.verifyToken(getTokenFromHeader(authHeader), false);
 
-        if(!idToken.equals(id)){
-            throw new UnauthorizedException();
-        }
+    return new ResponseEntity<>(
+        proprietaireService.getProprietaireById(id, isLogement),
+        HttpStatus.OK
+    );
+  }
 
-        return new ResponseEntity<>(
-                proprietaireService.updateProprietaire(id, proprietaireUpdateDTO),
-                HttpStatus.OK
-        );
+  /**
+   * Met à jour les informations d'un propriétaire.
+   *
+   * @param authHeader            L'en-tête d'autorisation contenant le token JWT.
+   * @param id                    L'identifiant du propriétaire à mettre à jour.
+   * @param proprietaireUpdateDTO Les données de mise à jour du propriétaire.
+   * @return Une ResponseEntity contenant le propriétaire mis à jour et le statut HTTP.
+   */
+  @PutMapping("/{id}")
+  public ResponseEntity<ProprietaireDTO> updateProprietaire(
+      @RequestHeader("Authorization") String authHeader, @PathVariable Long id,
+      @RequestBody ProprietaireUpdateDTO proprietaireUpdateDTO) {
+    System.out.println("test1");
+    Entry<Long, Boolean> tokenInfo = authService.verifyToken(getTokenFromHeader(authHeader), false);
+    System.out.println("test2");
+    if(!tokenInfo.getKey().equals(id) && !tokenInfo.getValue()){
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: 'if' is not followed by whitespace. Empty blocks                may only be represented as {} when not part of a multi-block statement (4.1.3)